### PR TITLE
[#2872] - add phone validation

### LINKED ISSUE #2872: No phone number validation on profile page
There seems to be no limits/validation for phone numbers on the profile page.  It seems like allowing the users to use characters like Ü could cause problems downstream in the system.  Additionally, analysis/parsing of user data might be messy when the phone number could take virtually any form.

[Video and Other Bug Report Files](https://github.com/hydroshare/hydroshare/files/2288939/003.zip)

### PR DESCRIPTION
**Pull request by soumyapurohit**
fixes #2872 

### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. [Enter positive test case here]

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|
| NEW DUPLICATED LINES DENSITY | 0.0 |
| NEW VULNERABILITIES | 0 |
| NEW BUGS | 0 |
| NEW CODE SMELLS | 19 |
| NEW SECURITY HOTSPOTS | 0 |

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4210](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4210)

File: theme/models.py [hunk 1]
```diff
@@ -266,13 +268,15 @@ class UserProfile(models.Model):
         blank=True,
         help_text="The name of the organization you work for."
     )
-    phone_1 = models.CharField(max_length=1024, null=True, blank=True)
+    phone_regex = RegexValidator(regex=r'^\d{8,15}$',
+                                 message="Phone number must be entered in the format: '999999999'. Up to 16 digits allowed.")
+    phone_1 = models.CharField(validators=[phone_regex], max_length=16, null=True, blank=True)
```
Comment (hyi) [inline theme/models.py:273]: I believe this requires a model migration. You can run ```docker exec -ti hydroshare python manage.py makemigrations``` if you have not already which should create a new migration file for this change. Then you can check in the migration file.

**Review by hyi — CHANGES_REQUESTED** [review]
@soumyapurohit It looks good, but you will need to check in the new migration file as well.

File: theme/models.py (hunk 1 above)
Comment (soumyapurohit) [reply theme/models.py:273]: Yes, it requires migration. Thanks for the command! I just checked in my file.

**Review by hyi — DISMISSED** [review]

**Comment by soumyapurohit** [comment]
retest this, please

**Review by hyi — APPROVED** [review]

**Pull request merged**